src/security-council-mgmt/SecurityCouncilManager.sol

```diff
@@ -0,0 +1,266 @@
+        }
+
+        // TODO: double check that this makes sense
+        bytes32 data = getRotateDataToSign(_currentAddress, _newAddress);
+        require(
+            _verify(data, _signature, _currentAddress), "SecurityCouncilManager: invalid signature"
+        );
+        for (uint256 i = 0; i < cohort.length; i++) {
+            if (cohort[i] == _currentAddress) {
+                cohort[i] = _newAddress;
+
+                address[] memory membersToAdd;
+                address[] memory membersToRemove;
+                membersToAdd[0] = _newAddress;
+                membersToRemove[0] = _currentAddress;
+                _dispatchUpdateMembers(membersToAdd, membersToRemove);
+            }
+        }
+        rotationNonce += 1;
+    }
+
+    /// @notice Get data to sign for a key rotation. Uses an incremented nonce
+    /// @param _currentAddress Address to rotate out
+    /// @param _newAddress Address to rotate in
+    function getRotateDataToSign(address _currentAddress, address _newAddress)
+        public
+        view
+        returns (bytes32)
+    {
+        return keccak256(abi.encodePacked(_currentAddress, _newAddress, rotationNonce));
+    }
+
+    /// @notice Verify signature over data
+    /// @param data Signed data
+    /// @param signature over data
+    /// @param account signer
+    /// @return true if signature is valid and from account
+    function _verify(bytes32 data, bytes memory signature, address account)
+        public
+        pure
+        returns (bool)
+    {
+        return data.toEthSignedMessageHash().recover(signature) == account;
+    }
+
+    /// @notice initates update to all Security Council Multisigs (gov chain, L1, and all others governed L2 chains).
+    /// Handles election results (add 6, remove 6 or fewer), add member (add one, remove none), and remove member (add none, remove one)
+    /// @param _membersToAdd array of members to add. can be empty.
+    /// @param _membersToRemove array of members to remove. can be empty.
+    function _dispatchUpdateMembers(
+        address[] memory _membersToAdd,
+        address[] memory _membersToRemove
+    ) internal {
+        // A candidate new be relected, which case they appear in both the arrays; removing and re-added is a no-op. We instead remove them from both arrays; this simplifies the logic of updating them in the gnosis safes.
+        (address[] memory newMembers, address[] memory oldMembers) =
+            SecurityCouncilMgmtUtils.removeSharedAddresses(_membersToAdd, _membersToRemove);
+        // Initiate L2 to L1 message to handle updating remaining secuirity councils
+        bytes memory data = abi.encodeWithSelector(
+            IL1SecurityCouncilUpdateRouter.handleUpdateMembers.selector, newMembers, oldMembers
+        );
+        ArbSys(0x0000000000000000000000000000000000000064).sendTxToL1(
+            l1SecurityCouncilUpdateRouter, data
+        );
```

Comment:
better to enforce a nonce in this message to prevent desynchronization

Reply:
with the new architecture enforcing ordering is harder; added a TODO / to discuss